airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/job.py

```diff
@@ -111,247 +109,209 @@ def job_size_adjusted_expand_factor(self, coef: float = 0.5) -> float:
-        raise ShopifyBulkExceptions.BulkJobUnknownError(f"Could not validate the status of the BULK Job `{self.job_id}`. Errors: {errors}.")
+    def _on_job_with_errors(self, errors: List[Mapping[str, Any]]) -> AirbyteTracedException:
+        raise ShopifyBulkExceptions.BulkJobUnknownError(
+            f"Could not validate the status of the BULK Job `{self._job_id}`. Errors: {errors}."
+        )
 
-    def job_check_for_errors(self, response: requests.Response) -> Union[AirbyteTracedException, Iterable[Mapping[str, Any]]]:
+    def _job_check_for_errors(self, response: requests.Response) -> Optional[Iterable[Mapping[str, Any]]]:
         try:
+
             return response.json().get("errors") or response.json().get("data", {}).get("bulkOperationRunQuery", {}).get("userErrors", [])
         except (Exception, JSONDecodeError) as e:
             raise ShopifyBulkExceptions.BulkJobBadResponse(
                 f"Couldn't check the `response` for `errors`, status: {response.status_code}, response: `{response.text}`. Trace: {repr(e)}."
             )
 
-    def job_one_time_retry_error(self, response: requests.Response, exception: Exception) -> Optional[requests.Response]:
-        if not self._one_time_error_retried:
-            request = response.request
-            self.logger.info(f"Stream: `{self.stream_name}`, retrying `Bad Request`: {request.body}. Error: {repr(exception)}.")
-            self._one_time_error_retried = True
-            return self.job_retry_request(request)
-        else:
-            self.on_job_with_errors(self.job_check_for_errors(response))
-
-    def job_track_running(self) -> Union[AirbyteTracedException, requests.Response]:
-        # format Job state check args
-        status_args = self.job_get_request_args(ShopifyBulkTemplates.status)
-        # re-use of `self._session(*, **)` to make BULK Job status checks
-        with self.session as track_running_job:
-            response = track_running_job.request(**status_args)
-        # errors check
-        try:
-            errors = self.job_check_for_errors(response)
-            if not errors:
-                self.job_update_state(response)
-                self.job_state_to_fn_map.get(self.job_state)(response=response)
-                return response
-            else:
-                # execute ERRORS scenario
-                self.on_job_with_errors(errors)
-        except (
-            ShopifyBulkExceptions.BulkJobBadResponse,
-            ShopifyBulkExceptions.BulkJobUnknownError,
-        ) as error:
-            return self.job_one_time_retry_error(response, error)
+    def _job_send_state_request(self) -> requests.Response:
+        with self.session as job_state_request:
+            status_args = self._job_get_request_args(ShopifyBulkTemplates.status)
+            self._request = requests.Request(**status_args, auth=self.session.auth).prepare()
+            return job_state_request.send(self._request)
```

Comment:
Should there be a try/except here in case of ConnectionError?

Reply:
We can probably wait until we have the CDK level of handler for the `ConnectionError`, and that would be covered out-of-the-box.